### PR TITLE
server: stop store when exit

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

File: src/server/server.rs [hunk 1]
```diff
@@ -566,8 +566,10 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Handler for Server<T, S> {
         // nothing to do now.
     }
 
-    fn interrupted(&mut self, event_loop: &mut EventLoop<Self>) {
-        event_loop.shutdown();
+    fn interrupted(&mut self, _: &mut EventLoop<Self>) {
```
Comment (ngaut) [inline src/server/server.rs]: Can we remove the method ?

File: src/server/server.rs [hunk 2]
```diff
@@ -566,8 +566,10 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Handler for Server<T, S> {
...
+        // To be able to be attached by gdb, we should not shutdown.
```
Comment (siddontang) [inline src/server/server.rs]: most of case, we won't use gdb for tikv, so I think catching `ctrl + c` and then closing is more convenient. 

Does `handle SIGINT stop` in gdb not work?

Comment (BusyJay) [inline src/server/server.rs]: I need gdb to get the stack trace when tikv is hung. `ctrl + c` should work too without implement this function.

File: src/server/server.rs (hunk 1 above)
Comment (BusyJay) [inline src/server/server.rs]: I propose to leave it here to remind implementing a gracefully shutdown.

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
@disksing PTAL

**Comment by disksing** [comment]
LGTM

**Pull request merged**